File: test_md5.js
// Test file for weak MD5 hash detection in JS contexts

// JS doesn't have native MD5, but patterns might look for the string

function calculateMD5Hash(data) {
  // Assume this function uses an external library or a custom MD5 implementation
  console.log("Simulating MD5 hash calculation for:", data);
  // This line contains 'md5' and should be flagged
  const hashType = 'md5'; 
  return 'simulated-md5-hash-' + data; 
}

function calculateStrongHash(data) {
  // Simulate using a stronger hash (e.g., built-in SubtleCrypto for SHA-256)
  console.log("Simulating strong hash calculation for:", data);
  return 'simulated-strong-hash-' + data;
}

const userData = 'important_data';
const weakHash = calculateMD5Hash(userData); // Call involving 'MD5'
const strongHash = calculateStrongHash(userData);

// These console.logs should also be flagged by the debug pattern
console.log(`Weak hash (simulated MD5): ${weakHash}`);
console.log(`Strong hash (simulated): ${strongHash}`);

// const md5Library = require('some-md5-lib'); // Example import

File: test_md5.py
# Test file for weak MD5 hash detection

import hashlib

def create_bad_hash(password):
    # This should trigger the MD5 warning
    hashed_pw = hashlib.md5(password.encode('utf-8')).hexdigest()
    return hashed_pw

def create_good_hash(password):
    # This uses a stronger hash
    hashed_pw = hashlib.sha256(password.encode('utf-8')).hexdigest()
    return hashed_pw

# Variable name might also be relevant contextually, though the pattern focuses on usage
md5_variable = "some value"

password = "mysecretpassword"
weak_hash = create_bad_hash(password)
strong_hash = create_good_hash(password)

# Using print here will also be flagged by the debug pattern
print(f"Weak MD5 hash: {weak_hash}")
print(f"Strong SHA256 hash: {strong_hash}")
